### PR TITLE
Integrate GolangCI-Lint

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Downstream the `golangci-lint` config from https://github.com/mattermost/mattermost-plugin-starter-template/pull/90 and sync `build`.

Most changes are straightforward and just style wise. I did comment on the more tricky ones.

#### Ticket Link
Follow up on https://github.com/mattermost/mattermost-plugin-autolink/pull/108

### CODE REVIEW

File: server/configuration_test.go [hunk 1]
```diff
@@ -73,10 +73,10 @@ func TestOnConfigurationChange(t *testing.T) {
 		"same configuration": {
 			SetupAPI: func() *plugintest.API {
 				api := &plugintest.API{}
-				api.On("LoadPluginConfiguration", mock.Anything).Return(nil)
-				api.On("GetTeams").Return([]*model.Team{&model.Team{Id: teamId}}, nil)
+				api.On("LoadPluginConfiguration", mock.AnythingOfType("*main.configuration")).Return(nil)
```
Comment (hanzei) [inline server/configuration_test.go:76]: Using `mock.AnythingOfType("*main.configuration")` is a bit more specific then `mock.Anything`

File: build/deploy/main.go [hunk 2]
```diff
@@ -85,13 +85,13 @@ func uploadPlugin(client *model.Client4, pluginID, bundlePath string) error {
...
 	if resp.Error != nil {
-		return fmt.Errorf("Failed to upload plugin bundle: %s", resp.Error.Error())
+		return errors.Wrap(resp.Error, "failed to upload plugin bundle")
 	}
 
 	log.Print("Enabling plugin.")
 	_, resp = client.EnablePlugin(pluginID)
 	if resp.Error != nil {
-		return fmt.Errorf("Failed to enable plugin: %s", resp.Error.Error())
+		return errors.Wrap(resp.Error, "Failed to enable plugin")
```
Comment (agarciamontoro) [inline build/deploy/main.go:94]: Are we enforcing `errors.Wrap` instead of `fmt.Errorf`?

File: server/configuration.go [hunk 3]
```diff
@@ -270,7 +270,10 @@ func (p *Plugin) ensureDemoUser(configuration *configuration) (string, error) {
 
 	for _, team := range teams {
 		// Ignore any error.
-		p.API.CreateTeamMember(team.Id, configuration.demoUserId)
+		_, err := p.API.CreateTeamMember(team.Id, configuration.demoUserID)
+		if err != nil {
+			p.API.LogError("Failed add demo user to team", "teamID", team.Id, "error", err.Error())
+		}
```
Comment (agarciamontoro) [inline server/configuration.go:276]: Shouldn't we remove the `//Ignore any error` comment?

File: server/http_hooks_test.go [hunk 4]
```diff
@@ -47,6 +47,7 @@ func TestServeHTTP(t *testing.T) {
 
 			result := w.Result()
 			require.NotNil(t, result)
+			defer result.Body.Close()
```
Comment (agarciamontoro) [inline server/http_hooks_test.go:50]: :+1:

**Comment by jfrerich** [comment]
@hanzei, I just ran `make check-style` on this branch and I actually see a failure.

image

**Comment by hanzei** [comment]
@jfrerich What does `golangci-lint version` say?

**Comment by jfrerich** [comment]
I've got version 1.25.1.  It was released yesterday https://github.com/golangci/golangci-lint/releases

File: build/deploy/main.go (hunk 2 above)
Comment (hanzei) [reply build/deploy/main.go:94]: Not, we are not enforcing this, but you are bringing up another great points.

The complain from `golangci-lint` for the original message is:
` error strings should not be capitalized or end with punctuation or a newline (golint)`, which is actually not fixed with the change. `golangci-lint` just doesn't find the capitalized message.

But given that `build` is just a copy from https://github.com/mattermost/mattermost-plugin-starter-template I leaning towards not fixing it in this PR.

**Comment by hanzei** [comment]
@jfrerich I'm fairly confident that the failure you are reporting was introduced with https://github.com/golangci/golangci-lint/commit/dfb22785dc8262d3bfe7ff461938ef2464d3cbe9, which is part of `v1.25.1` but not `v1.25.0`. Given that CI is using `v1.24.0` and the code in question is part of https://github.com/mattermost/mattermost-plugin-starter-template, I'm leaning toward ignoring the error in this PR and fixing it in https://github.com/mattermost/mattermost-plugin-starter-template.

**Comment by jfrerich** [comment]
@hanzei, I agree that ignoring is ok if fine.  I've added the `//nolint` comment to ignore the bookmarks plugin repo.

**Comment by hanzei** [comment]
@jfrerich Would you please report the gosec issue on https://github.com/mattermost/mattermost-plugin-starter-template?

**Comment by jfrerich** [comment]
@hanzei.  Done.

**Comment by hanzei** [comment]
@jfrerich Is there anything that needs to be done in this PR before you approve it?

**Comment by hanzei** [comment]
@DHaussermann This is the last PR that should go into `v0.7.0`. Maybe it's fine to merge it and do proper testing as part of https://github.com/mattermost/mattermost-plugin-demo/pull/106?

**Pull request merged**